demisto_sdk/commands/test_content/TestContentClasses.py

```diff
@@ -0,0 +1,1639 @@
+        self.is_nightly = kwargs['nightly']
+        self.slack_client = SlackClient(kwargs['slack'])
+        self.circleci_token = kwargs['circleci']
+        self.build_number = kwargs['build_number']
+        self.build_name = kwargs['build_name']
+        self.isAMI = kwargs['is_ami']
+        self.memCheck = kwargs['mem_check']
+        self.server_version = kwargs['server_version']
+        self.specific_tests_to_run = self._parse_tests_list_arg(kwargs['tests_list'])
+        self.is_local_run = (self.server is not None)
+        self.server_numeric_version = self._get_server_numeric_version()
+        self.instances_ips = self._get_instances_ips()
+        self.filtered_tests = self._extract_filtered_tests()
+        self.tests_data_keeper = TestResults(self.conf.unmockable_integrations)
+        self.conf_unmockable_tests = self._get_unmockable_tests_from_conf()
+        self.unmockable_test_ids: Set[str] = set()
+        self.mockable_tests_to_run, self.unmockable_tests_to_run = self._get_tests_to_run()
+        self.slack_user_id = self._retrieve_slack_user_id()
+        self.all_integrations_configurations = self._get_all_integration_config(self.instances_ips[0])
+
+    def _get_all_integration_config(self, server_ip: str) -> Optional[list]:
+        """
+        Gets all integration configuration as it exists on the demisto server
+        Since in all packs are installed the data returned from this request is very heavy and we want to avoid
+        running it in multiple threads.
+        Args:
+            server_ip: The url of the server to create integration in
+
+        Returns:
+            A dict containing the configuration for the integration if found, else None
+        """
+        if not self.is_nightly:
+            return []
+        server_url = f'https://{server_ip}'
+        tmp_client = demisto_client.configure(base_url=server_url,
+                                              api_key=self.api_key,
+                                              verify_ssl=False)
+        self.logging_module.debug('Getting all integrations instances')
+        try:
+            res_raw = demisto_client.generic_request_func(self=tmp_client, path='/settings/integration/search',
+                                                          method='POST', body={})
+        except ApiException:
+            self.logging_module.exception('failed to get all integrations configuration')
+            return None
+
+        res = ast.literal_eval(res_raw[0])
+        TIMEOUT = 180
+        SLEEP_INTERVAL = 5
+        total_sleep = 0
+        while 'configurations' not in res:
+            if total_sleep == TIMEOUT:
+                self.logging_module.error(
+                    f"Timeout - failed to get all integration configuration. Error: {res}")
+                return None
+
+            time.sleep(SLEEP_INTERVAL)
+            total_sleep += SLEEP_INTERVAL
+
+        all_configurations = res['configurations']
```

Comment:
Seems you have code duplication on this - you might want to move this to a tool.